src/main/java/org/broadinstitute/hellbender/engine/spark/SparkCommandLineProgram.java

```diff
@@ -85,4 +93,36 @@ protected void afterPipeline(final JavaSparkContext ctx) {
     protected String getProgramName(){
         return programName == null ? getClass().getSimpleName() : programName;
     }
+
+    /**
+     * Returns the number of cores being used by each worker.
+     * In local mode it will return the total number of cores allocated to the process.
+     */
+    protected int getExecutorCores() {
+        int cores;
+        SchedulerBackend schedulerBackend = ctx.sc().schedulerBackend();
+        if (schedulerBackend instanceof LocalBackend) {
+            cores = ((LocalBackend) schedulerBackend).totalCores();
+        } else {
+            cores = ctx.getConf().getInt("spark.executor.cores",1); // Spark defaults to 1 core per executor if no value is set
+        }
+        return cores;
+    }
+
+    /**
+     * Returns the total memory allocated to each core (in MB)
+     *
+     * NOTE: in local mode this returns the ammount of memory available to the primary executor for caching
+     * by default this value will be approximately 0.5 of the available memory for one core. Additionally,
+     * there is no guarantee that this result will be correct if the user never specifies either 'spark.executor.memory'
+     * or 'spark.executor.cores'
+     */
+    public double getMemoryPerCore() {
+        if (ctx.isLocal()) {
+            // In local mode, poll the active executor directly for the memory it has
+            return ((long)ctx.sc().getExecutorMemoryStatus().valuesIterator().next()._1() / ((double)getExecutorCores()))/(1000000.0);
+        }
+        return ctx.sc().executorMemory() / ((double)getExecutorCores());
```

Comment:
The method `executorMemory()` has also become undocumented as of Spark 1.4.  But in any case, this returns total executor memory in MB, without subtracting reserve and user memory.  Whereas `getExecutorMemoryStatus()` returns just the Spark memory (cache and storage).  So the two methods aren't commensurate.
I'd like to see us use `getExecutorMemoryStatus()` both locally and on the cluster.  There are two problems, however:  There's a race condition where if you call it too quickly on a brand new JavaSparkContext when on a real cluster, it gives a bogus answer (it just reports the driver).  The second problem is that it's not clear how to identify which Map entry is for the driver and which entries are for nodes.  But if we can figure out how to do that, we could return a nice report that gave nicely complete information:  driver memory, executor spark memory per core, executor user memory per core, number of executors, number of cores per executor.  A pretty complete description of the execution environment.  This would be really valuable.
Can you figure out how to identify the driver?